lib/promscrape/scrapework.go

```diff
@@ -350,9 +351,15 @@ func (sw *scrapeWork) run(stopCh <-chan struct{}, globalStopCh <-chan struct{})
 			return
 		case tt := <-ticker.C:
 			t := tt.UnixNano() / 1e6
-			if d := math.Abs(float64(t - timestamp)); d > 0 && d/float64(scrapeInterval.Milliseconds()) > 0.1 {
-				// Too big jitter. Adjust timestamp
-				timestamp = t
+			if lag := math.Abs(float64(t - timestamp)); lag > 0 {
+				si := float64(scrapeInterval.Milliseconds())
+				if lag/si > 0.1 {
+					// Too big jitter. Adjust timestamp
+					timestamp = t
+				}
+				if lag/si >= 1 {
+					scrapesSkipped.Inc()
```

Comment:
If I read this correctly, we will only increase metric by 1 even if multiple scrapes were skipped, maybe it will be better to increase metric by `lag/si`?